launchdarkly-android-client-sdk/build.gradle

```diff
@@ -138,19 +138,6 @@ task javadocJar(type: Jar, dependsOn: javadoc) {
     from javadoc.destinationDir
 }
 
-// Force the Javadoc build to fail if there are any Javadoc warnings. See: https://discuss.gradle.org/t/javadoc-fail-on-warning/18141/3
-if (JavaVersion.current().isJava8Compatible()) {
-    tasks.withType(Javadoc) {
-        // The '-quiet' as second argument is actually a hack,
-        // since the one parameter addStringOption doesn't seem to
-        // work, we extra add '-quiet', which is added anyway by
-        // gradle. See https://github.com/gradle/gradle/issues/2354
-        // See JDK-8200363 (https://bugs.openjdk.java.net/browse/JDK-8200363)
-        // for information about the -Xwerror option.
-        options.addStringOption("Xwerror", "-quiet")
-    }
-}
-
```

Comment:
After these action changes, this "treat warnings as errors" step started failing the build.  I'm unsure why this was not failing before when using Circle CI.

I fixed as many of the warnings as I could, but could not resolve warnings on autogenerated files such as BuildConfig.java.  After about 3 hours of investigating, decided to eliminate this to move on to other priorities.  Ultimately this PR does fix ~80 out of the ~85 warnings that it was giving.

Reply:
Ok. We should probably file a task to come back to it. Ideally we could ignore those generated files, but I know that tooling isn't always designed with people actually using it in mind.

Reply:
How about lumping it in with the mono repo work?

Reply:
Created SC-235492 to track this